pkg/storage/ec/client.go

```diff
@@ -126,6 +126,16 @@ func (ec *ecClient) Put(ctx context.Context, nodes []*pb.Node, rs eestream.Redun
 		}
 	}
 
+	/* clean up the partially uploaded segment's pieces */
+	defer func() {
+		select {
+		case <-ctx.Done():
+			ctx = context.Background()
+			_ = ec.Delete(ctx, successfulNodes, pieceID)
```

Comment:
Another issue on this line is that we try to delete the pieces only for the `successfulNodes`. This is the reason for leaking some pieces when cancelling the upload. It happens because we rely on the piece store client do delete the partially uploaded pieces when it gets `io.ErrExpectedEOF`, but some of these attempts fail because the context is already cancelled: https://github.com/storj/storj/blob/master/pkg/piecestore/rpc/client/client.go#L132

The easiest way to resolve this is to try to delete from all `nodes` instead of just `successfulNodes`, i.e. the call should be like:

```go
ec.Delete(ctx.Background(), nodes, pieceID)
```

Reply:
done